### DISCUSSION TITLE
Improving GPU-Z

### DISCUSSION THREAD

**Discussion by W1zzardTPU** [xCuri0/ReBarUEFI#132] [Q&A]
Any suggestions how detection of CSM / 4G Decoding can be improved in GPU-Z?

**Comment by xCuri0** [comment]
Are you able to explain how the current method for detecting 4G decoding works ? I've found what works for 4G decode detection is seeing if large memory is visible in device manager resources.

Also idk if this is fixed now but someone reported that GPU-Z won't detect 64GB BAR size correctly https://github.com/xCuri0/ReBarUEFI/issues/23#issuecomment-1360803086

Someone else had a question about how GPU-Z detects resizable BAR differently from AMD drivers, I also encountered the same thing when I manipulate BAR size after POST (but before Windows bootloader) https://github.com/xCuri0/ReBarUEFI/discussions/118#discussion-6101502. I couldn't figure out any possible reason for this so hoping to get an answer.

**Comment by W1zzardTPU** [comment]
> Are you able to explain how the current method for detecting 4G decoding works ? I've found what works for 4G decode detection is seeing if large memory is visible in device manager resources.

I think that's exactly what I do. I first check if the GPU device has any BAR address>4G, if not, then ask Device Manager for all devices on any PCI bus and check if any one of those has a BAR>4G.

> Someone else

Very sad that these people don't just submit a bug report to me, so I can fix, instead of everyone trying to mess around to get a result they hope for
